x/trigger/simulation/operations.go

```diff
@@ -0,0 +1,171 @@
+package simulation
+
+import (
+	"math/rand"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/simapp/helpers"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/cosmos/cosmos-sdk/x/bank/testutil"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	simappparams "github.com/provenance-io/provenance/app/params"
+	"github.com/provenance-io/provenance/internal/pioconfig"
+	"github.com/provenance-io/provenance/x/trigger/keeper"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+// Simulation operation weights constants
+const (
+	//nolint:gosec // not credentials
+	OpWeightMsgCreateTrigger = "op_weight_msg_create_trigger"
+	//nolint:gosec // not credentials
+	OpWeightMsgDestroyTrigger = "op_weight_msg_destroy_trigger"
+)
+
+// WeightedOperations returns all the operations from the module with their respective weights
+func WeightedOperations(
+	appParams simtypes.AppParams, cdc codec.JSONCodec, k keeper.Keeper, ak authkeeper.AccountKeeperI, bk bankkeeper.Keeper,
+) simulation.WeightedOperations {
+	var (
+		weightMsgCreateTrigger  int
+		weightMsgDestroyTrigger int
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgCreateTrigger, &weightMsgCreateTrigger, nil,
+		func(_ *rand.Rand) {
+			weightMsgCreateTrigger = simappparams.DefaultWeightSubmitCreateTrigger
+		},
+	)
+	appParams.GetOrGenerate(cdc, OpWeightMsgDestroyTrigger, &weightMsgDestroyTrigger, nil,
+		func(_ *rand.Rand) {
+			weightMsgDestroyTrigger = simappparams.DefaultWeightSubmitDestroyTrigger
+		},
+	)
+
+	return simulation.WeightedOperations{
+		simulation.NewWeightedOperation(
+			weightMsgCreateTrigger,
+			SimulateMsgCreateTrigger(k, ak, bk),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgDestroyTrigger,
+			SimulateMsgDestroyTrigger(k, ak, bk),
+		),
+	}
+}
+
+// SimulateMsgCreateTrigger sends a MsgCreateTriggerRequest.
+func SimulateMsgCreateTrigger(_ keeper.Keeper, ak authkeeper.AccountKeeperI, bk bankkeeper.Keeper) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		now := ctx.BlockTime()
+		from, _ := simtypes.RandomAcc(r, accs)
+		to, _ := simtypes.RandomAcc(r, accs)
+		for to.Equals(from) {
+			to, _ = simtypes.RandomAcc(r, accs)
+		}
```

Comment:
Not good.

If there's only one account, this is an infinite loop. If there's two accounts, this can still loop more than anyone would ever expect.

Also, the `_` coming back from `RandomAcc` is the index of the entry. It'd be easier to just compare the indexes.

If you update `GetRandomAction` to take in `accs` though, you can have it pick the to and from. You'd probably want it to also return the `from` `simtypes.Account` though since it's needed for `Dispatch`.

Otherwise, it might be worthwhile to create a `RandomAccs` function that looks something like this:
```golang
func RandomAccs(r *rand.Rand, accs []simtypes.Account, count int) ([]simtypes.Account, error) {
	if len(accs) < count {
		return fmt.Errorf("cannot choose %d accounts because there are only %d", count, len(accs)
	}
	raccs := make([]simtypes.Account, 0, len(accs)
	raccs = append(raccs, accs...)
	r.Shuffle(len(raccs), func(i, j int) {
		raccs[i], raccs[j] = raccs[j], raccs[i]
	})
	return raccs[:count]
}